app/routes.py

```diff
@@ -1,2 +1,251 @@
+def get_one_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    if not customer:
+        return ({"error": f"No customer with ID #{customer_id}"}, 404)
+    return customer.to_dict()
+
+
+@customers_bp.route("", methods=["POST"])
+def create_customer():
+    request_body = request.get_json()
+    valid_customer_input(request_body)
+    new_customer = Customer(name = request_body["name"],
+                    postal_code = request_body["postal_code"],
+                    phone = request_body["phone"])
+    new_customer.registered_at = datetime.utcnow()
+    db.session.add(new_customer)
+    db.session.commit()
+    return ({"id": new_customer.customer_id}, 201) 
+
+
+@customers_bp.route("/<customer_id>", methods=["PUT"])
+def update_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    if not customer:
+        return ({"error": f"No customer with ID #{customer_id}"}, 404)
+    form_data = request.get_json()
+    valid_customer_input(form_data)
+    customer.name = form_data["name"]
+    customer.postal_code = form_data["postal_code"]
+    customer.phone = form_data["phone"]
+    db.session.commit()
+    return customer.to_dict()
+    
+
+@customers_bp.route("/<customer_id>", methods=["DELETE"])
+def delete_customer(customer_id):
+    customer = Customer.query.get(customer_id)
+    if not customer:
+        return ({"error": f"No customer with ID #{customer_id}"}, 404)
+    db.session.delete(customer)
+    db.session.commit()
+    return make_response({"id": customer.customer_id}, 200)
+
+
+@videos_bp.route("", methods=["GET"])
+def get_videos():
+    videos = Video.query.all()
+    videos_response = [video.to_dict() for video in videos]
+    return jsonify(videos_response)
+
+
+@videos_bp.route("/<video_id>", methods=["GET"])
+def get_one_video(video_id):
+    video = Video.query.get(video_id)
+    if not video:
+        return ({"error": f"No video with ID #{video_id}"}, 404)
+    return video.to_dict()
+
+
+@videos_bp.route("", methods=["POST"])
+def create_video():
+    request_body = request.get_json()
+    valid_video_input(request_body)
+    new_video = Video(title = request_body["title"],
+                    release_date = request_body["release_date"],
+                    total_inventory = request_body["total_inventory"])
+    new_video.available_inventory = new_video.total_inventory
+    db.session.add(new_video)
+    db.session.commit()
+    return ({"id": new_video.video_id}, 201) 
+
+
+@videos_bp.route("/<video_id>", methods=["PUT"])
+def update_video(video_id):
+    video = Video.query.get(video_id)
+    if not video:
+        return ({"error": f"No video with ID #{video_id}"}, 404)
+    form_data = request.get_json()
+    valid_video_input(form_data)
+    video.title = form_data["title"]
+    video.release_date = form_data["release_date"]
+    video.total_inventory = form_data["total_inventory"]
+    #new line below
+    video.available_inventory = form_data["available_inventory"]
```

Comment:
Since you use `[]` without first checking if a key exists (this one isn't in `valid_video_input`) you can return a 500 here from raising an exception if the user doesn't provide `available_inventory` (like one of the Postman tests does):

You can use `get` to make your endpoint more resilient:
```suggestion
    video.available_inventory = form_data.get("available_inventory")
```